### ISSUE TITLE
How to let Learning rate be first linearly increased from 0 to 0.16 in the first training epoch and then annealed down using cosine decay rule.

### ISSUE THREAD

**Issue by Li505358678** [google/automl#205]
I would like to reproduce the experimental content of the paper based on coco2017. The training result is not ideal. It may be the problem of learning rate. How to modify the code

**Comment by Li505358678** [comment]
I see. I find that learning rate is related to batch size

**Comment by Li505358678** [comment]
My batch-size is 4，I change the default_batch_size to 2，train on coco2017，use instances_train2017.json,but display NAN . Is learning rate wrong? I just do as paper said

**Comment by Li505358678** [comment]
@mingxingtan How do you train the model? Why do I have so many problems according to the paper,

**Comment by mingxingtan** [comment]
As stated in the paper, I used 32 TPUv3 cores, with total batch size 128  (4 per core).

Are you using single GPU? If your total batch size is only 2, batch normalization might be unstable and thus causes some issues during training.  Considering some people don't have many GPUs, we may need to replace batch norm with group norm or other batch-size invariant normalizations in future.

Are you interested in this task?

**Comment by Li505358678** [comment]
> As stated in the paper, I used 32 TPUv3 cores, with total batch size 128 (4 per core).
> 
> Are you using single GPU? If your total batch size is only 2, batch normalization might be unstable and thus causes some issues during training. Considering some people don't have many GPUs, we may need to replace batch norm with group norm or other batch-size invariant normalizations in future.
> 
> Are you interested in this task?

I‘m sorry，I don't know much about this area，so I can’t help you.

I use single GPU with 16G，so the train_batch_size’s maximum is 8, I remember that this code can't be trained with multi GPU. Is it ok now?and What size GPU can run it with batch size 128

What's more ,I don't see Training data processing in the readme , is it trained just like val data processing? I use caption_annotation_file by caption_train2017.json when I processed train data.  the box_loss always be 0,  so I changed to use object_annotations_file by instance_train2017.json. How do you do it? I saw the readme in dataset folder, the best term is image_info_file. Now I can't see about it and I don't know how to deal with this use caption?or object?

If I change  learning rate to be first linearly increased from 0.008(as your code but your paper is about 0) to 0.016 or  lower in the first training epoch and then annealed down using cosine decay rule. the result of efficientdet-d0 will be low just 
![image](https://user-images.githubusercontent.com/51314795/79098360-60bf4100-7d94-11ea-97fc-19c8647465d6.png)

**Comment by Li505358678** [comment]
I now resize image of train2017 to 0.5 times  and change the json in order to enlarge the batch size ,Is that ok? Is there any problem?

**Comment by Li505358678** [comment]
after resize,batch size can only be 16,it is small too

**Comment by Li505358678** [comment]
Unbelievable，I trained with 32 GPU ,the batch-size can only be 32 that is quarter of the it in paper. I can't think about that how many GPU needed with batch size 128

**Comment by mingxingtan** [comment]
Related to #459 459
